bootstrap/http/src/main/kotlin/com/studentcenter/weave/bootstrap/chat/controller/ChatMessageEventHandler.kt

```diff
@@ -0,0 +1,57 @@
+package com.studentcenter.weave.bootstrap.chat.controller
+
+import com.studentcenter.weave.application.chat.port.inbound.SaveChatMessage
+import com.studentcenter.weave.domain.chat.entity.ChatMessage
+import com.studentcenter.weave.domain.chat.event.ChatMessageConsumeEvent
+import io.github.oshai.kotlinlogging.KotlinLogging
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.springframework.context.event.EventListener
+import org.springframework.messaging.simp.SimpMessageSendingOperations
+import org.springframework.stereotype.Controller
+
+@Controller
+class ChatMessageEventHandler(
+    private val simpMessageSendingOperations: SimpMessageSendingOperations,
+    private val saveChatMessage: SaveChatMessage,
+) {
+
+    private val logger = KotlinLogging.logger { this::class.java.name }
+
+    @EventListener
+    fun handleChatMessageConsumeEvent(event: ChatMessageConsumeEvent) {
+        val chatMessage: ChatMessage = event.entity
+
+        CoroutineScope(Dispatchers.IO).launch {
+            val broadCastJob: Deferred<Unit> = async { broadCastChatMessage(chatMessage) }
+            val saveJob = async { saveChatMessage.invoke(chatMessage) }
```

Comment:
p3) 이 부분 메세지만 넘겨주고 캐싱 및 로깅 호출하는 하나의 메서드로 만들어서 바로 호출하면 좋을 것 같은데 어떤가요? 
아래와 같은 느낌으로?
```suggestion
            async { broadCastChatMessage(chatMessage) }.CatchAndFailureLogging("broadcast chat meesage") 
            async { saveChatMessage.invoke(chatMessage) }.CatchAndFailureLogging("save chat meesage") 
```

Reply:
await 때문에 저렇게 체이닝가능할지 모르겠네요 일단 코루틴 연구좀 해볼게용 ㅋㅋ